utils/prometheus-node-exporter-lua/files/usr/bin/prometheus-node-exporter-lua

```diff
@@ -0,0 +1,391 @@
+  for i, e in ipairs(natstat) do
+    -- output(string.format("%s\n",e  ))
+    local fields = space_split(e)
+    local src, dest, bytes;
+    bytes = 0;
+    for ii, field in ipairs(fields) do
+      if src == nil and string.match(field, '^src') then
+        src = string.match(field,"src=([^ ]+)");
+      elseif dest == nil and string.match(field, '^dst') then
+        dest = string.match(field,"dst=([^ ]+)");
+      elseif string.match(field, '^bytes') then
+        local b = string.match(field, "bytes=([^ ]+)");
+        bytes = bytes + b;
+        -- output(string.format("\t%d %s",ii,field  ));
+      end
+
+    end
+    -- local src, dest, bytes = string.match(natstat[i], "src=([^ ]+) dst=([^ ]+) .- bytes=([^ ]+)");
+    -- local src, dest, bytes = string.match(natstat[i], "src=([^ ]+) dst=([^ ]+) sport=[^ ]+ dport=[^ ]+ packets=[^ ]+ bytes=([^ ]+)")
+
+    local labels = { src = src, dest = dest }
+    -- output(string.format("src=|%s| dest=|%s| bytes=|%s|", src, dest, bytes  ))
+    nat_metric(labels, bytes )
+  end
+end
+
+function timed_scrape(scraper)
+  local start_time = socket.gettime()
+  -- build the function name and call it from global variable table
+  _G["scraper_"..scraper]()
+  local duration = socket.gettime() - start_time
+  return duration
+end
+
+function run_all_scrapers()
+  times = {}
+  for i,scraper in ipairs(scrapers) do
+    runtime = timed_scrape(scraper)
+    times[scraper] = runtime
+    scrape_time_sums[scraper] = scrape_time_sums[scraper] + runtime
+    scrape_counts[scraper] = scrape_counts[scraper] + 1
+  end
+
+  local name = "node_exporter_scrape_duration_seconds"
+  local duration_metric = metric(name, "summary")
+  for i,scraper in ipairs(scrapers) do
+    local labels = {collector=scraper, result="success"}
+    duration_metric(labels, times[scraper])
+    print_metric(name.."_sum", labels, scrape_time_sums[scraper])
+    print_metric(name.."_count", labels, scrape_counts[scraper])
+  end
+end
+
+-- Web server-specific functions
+
+function http_ok_header()
+  output("HTTP/1.1 200 OK\r")
+  output("Server: lua-metrics\r")
+  output("Content-Type: text/plain; version=0.0.4\r")
+  output("\r")
+end
+
+function http_not_found()
+  output("HTTP/1.1 404 Not Found\r")
+  output("Server: lua-metrics\r")
+  output("Content-Type: text/plain\r")
+  output("\r")
+  output("ERROR: File Not Found.")
+end
+
+function serve(request)
+  if not string.match(request, "GET /metrics.*") then
+    http_not_found()
+  else
+    http_ok_header()
+    run_all_scrapers()
+  end
+  client:close()
+  return true
+end
+
+-- Main program
+
+for k,v in ipairs(arg) do
+  if (v == "-p") or (v == "--port") then
+    port = arg[k+1]
+  end
+  if (v == "-b") or (v == "--bind") then
+    bind = arg[k+1]
+  end
+end
+
+scrape_counts = {}
+scrape_time_sums = {}
+for i,scraper in ipairs(scrapers) do
+  scrape_counts[scraper] = 0
+  scrape_time_sums[scraper] = 0
+end
+
+if port then
+  server = assert(socket.bind("*", port))
```

Comment:
I just forgot to use the cli argument from here:

https://github.com/simonswine/openwrt-packages/blob/ccdf9c047ccd32ab427b012a198cef5372656856/utils/prometheus-node-exporter-lua/files/usr/bin/prometheus-node-exporter-lua#L362

 :see_no_evil:, will fix later...